src/storage/FileLoader.cpp

```diff
@@ -57,17 +79,26 @@ static std::string getKey(const faabric::Message& msg,
     return key;
 }
 
-static boost::filesystem::path getDir(std::string baseDir,
-                                      const faabric::Message& msg,
-                                      bool create)
+static std::filesystem::path getDir(std::string baseDir,
+                                    const faabric::Message& msg,
+                                    bool create)
 {
-    boost::filesystem::path path(baseDir);
+    std::filesystem::path path(baseDir);
     path.append(msg.user());
     path.append(msg.function());
 
     // Create directory if required
     if (create) {
-        boost::filesystem::create_directories(path);
+        try {
+            std::filesystem::create_directories(path);
+        } catch (const std::filesystem::filesystem_error& ex) {
+            SPDLOG_ERROR(
+              "Call to create_directories ({}) failed with error ({}): {}",
+              path.string(),
+              ex.code().value(),
+              ex.what());
+            throw std::runtime_error("Filesystem error creating directories");
+        }
```

Comment:
This snippet of code calling `create_directories` is repeated three times. It should be in a function called something like `createDirectories` which can just be `static` in this file for now.